### ISSUE TITLE
anomaly score

### ISSUE THREAD

**Issue by lzzlxxlsz** [LeeDoYup/AnoGAN#19]
When I finished the anomaly test stage with the anomal score,and how can I know the anomaly picture

**Comment by LeeDoYup** [comment]
In this paper, the detail method is not presented.
However, the basic approach is construct validation dataset and set anomaly threshold based on validation sets.

**Comment by lzzlxxlsz** [comment]
Thank you! That means I have to set a threshold to determine whether a picture is abnormal or not.With your test_data_01.png ,the anomal score is 3185.73901367,is result biggger？Thanks a lot !

**Comment by LeeDoYup** [comment]
There is anomaly score in each pixel and the value is the aggregation of all the pixel.
So it can be large.
Rather than considering absolute value, you can define the threshold based on validation detection accuracy. (That is, you need labeled-anomaly-data for validation)

**Comment by lzzlxxlsz** [comment]
That is to say,I use the validation dataset(labeled) for testing,then find a better threshold

**Comment by lzzlxxlsz** [comment]
There is another question ，what is the anomal situation for the celebA dataset？is there a hat or glasses and be the anomal ?

**Comment by pradeepbhaskar** [comment]
Hi Lee,

Could you please elaborate what does "labeled-anomaly-data" mean.
When I am testing my model using anomaly image, I get 2 images on the "test_results" directory. 
First image file is all blank and other mage file ( which has "error" in its name) is the gray scale image of the anomaly image. A bit confused as to how to interpret. 



--------------------------------------------------------------------------------
Epoch: [97], anomaly score: 71413.89062500, res loss: 68573.62500000
Epoch: [98], anomaly score: 71413.28906250, res loss: 68573.62500000
Epoch: [99], anomaly score: 71412.67187500, res loss: 68573.60937500
--------------------------------------------------------------------------------
./test_data/test_result/AD_0_IMG_20191020_120417.jpg
./test_data/test_result/AD_50_IMG_20191020_120417.jpg
./test_data/test_result/AD_99_IMG_20191020_120417.jpg

./test_data/test_result/AD_error_0_IMG_20191020_120417.jpg
./test_data/test_result/AD_error_50_IMG_20191020_120417.jpg
./test_data/test_result/AD_error_99_IMG_20191020_120417.jpg

Right now I am training with 100 epochs, should I train more ? 
My test epoch is 100 , should it be increased ? Please assist.

thanks,
Pradeep

**Comment by sachitha-bandara** [comment]
@pradeepbhaskar were you able to find the answers for the above question? @LeeDoYup I am also confused with the images saved when the test is running. Please elaborate. Thanks in advance.
